src/v/pandaproxy/schema_registry/test/post_subjects_subject_version.cc

```diff
@@ -105,6 +109,337 @@ FIXTURE_TEST(
+  "version": 3,
+  "schemaType": "AVRO"
+})"};
+
+    pps::subject subject{"in-version-order"};
+    put_config(client, subject, pps::compatibility_level::backward);
+
+    {
+        info("Post schema v2");
+        auto res = post_schema(client, subject, schema_v2);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        res = get_subject_versions(client, subject);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        std::vector<pps::schema_version> expected{pps::schema_version{2}};
+        auto versions = get_body_versions(res.body);
+        BOOST_REQUIRE_EQUAL(versions, expected);
+    }
+
+    {
+        info("Post schema v3_no_default (expect conflict)");
+        auto res = post_schema(client, subject, schema_v3_no_default);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::conflict);
+    }
+
+    {
+        info("Post schema v3");
+        auto res = post_schema(client, subject, schema_v3);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        res = get_subject_versions(client, subject);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        std::vector<pps::schema_version> expected{
+          pps::schema_version{2}, pps::schema_version{3}};
+        auto versions = get_body_versions(res.body);
+        BOOST_REQUIRE_EQUAL(versions, expected);
+    }
+
+    subject = pps::subject{"in-reverse-order"};
+    put_config(client, subject, pps::compatibility_level::backward);
+
+    {
+        info("Post schema v3_no_default");
+        auto res = post_schema(client, subject, schema_v3_no_default);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        res = get_subject_versions(client, subject);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        std::vector<pps::schema_version> expected{pps::schema_version{3}};
+        auto versions = get_body_versions(res.body);
+        BOOST_REQUIRE_EQUAL(versions, expected);
+    }
+
+    {
+        info("Post schema v2 (expect success, despite incompatibility)");
+        auto res = post_schema(client, subject, schema_v2);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        res = get_subject_versions(client, subject);
+        BOOST_REQUIRE_EQUAL(
+          res.headers.result(), boost::beast::http::status::ok);
+
+        std::vector<pps::schema_version> expected{
+          pps::schema_version{2}, pps::schema_version{3}};
+        auto versions = get_body_versions(res.body);
+        BOOST_REQUIRE_EQUAL(versions, expected);
+    }
```

Comment:
I am confused why it is OK for this request to succeed if schema_v3_no_default and schema_v2 or in-compatible. Should there be a compatibility check somewhere within the SR?